### PR TITLE
Query object translation with only manage_translation permission

### LINKED ISSUE #4834: Translate eg. products without manage_products permissions
### What I'm trying to achieve
To translate hidden products, sales and other permission-protected objects.

### Describe a proposed solution
```
translation(kind: PRODUCT, id: "", languageCode: DE) {
  ..on Product {
    id
    name
    description
    ...
    translation(languageCode: $languageCode) {
      id
      name
      description
      ...
    }
  }
}
```

### PR DESCRIPTION
**Pull request by fowczarek**
I want to merge this change because resolve #4834 

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [ ] The changes are tested.
1. [ ] GraphQL schema and type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by fowczarek** [comment]
@dominik-zeglen Please check if you have all needed information in this query for Product translation.

**Comment by django-queries** [comment]
Here is the report for 8c3cda9774897c09723db80d1b833866164d61f1 (mirumee/saleor @ 4834/refactor_translations_query)
Base comparison is 041f2aeab90b132abbbde9c505355413c3e4d601.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  add billing address to checkout            	         34	         34	             20
  add shipping to checkout                   	          7	          7	              0
  checkout payment charge                    	         14	         14	              0
  complete checkout                          	          6	          6	              0
  create checkout                            	         48	         48	             24

# api.benchmark homepage
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  retrieve main menu                         	          5	          5	              0
  retrieve product list                      	          4	          4	              0
  retrieve secondary menu                    	          5	          5	              0
  retrieve shop                              	          2	          2	              0

# api.benchmark product
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product details                            	         15	         15	              3
  retrieve product attributes                	         13	         13	              2

# api.benchmark variant
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product variant bulk create                	         51	         51	              3
  retrieve variant list                      	         15	         15	              6

# api product sorting attributes
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  sort product not having attribute data     	         21	         21	              0
```

</p>
</details>

File: saleor/graphql/translations/schema.py [hunk 1]
```diff
@@ -88,3 +105,10 @@ def resolve_translations(self, info, kind, **_kwargs):
             return resolve_menu_items(info, query=None)
         elif kind == TranslatableKinds.SALE:
             return resolve_sales(info, query=None)
+
+    @permission_required("site.manage_translations")
+    def resolve_translation(self, info, id, kind, **_kwargs):
+        if kind == TranslatableKinds.PRODUCT:
+            _type, product_id = graphene.Node.from_global_id(id)
+            return Product.objects.get(pk=product_id)
```
Comment (korycins) [inline saleor/graphql/translations/schema.py]: This will fail. Use filter.first

File: saleor/graphql/page/resolvers.py [hunk 2]
```diff
@@ -2,29 +2,19 @@
...
 from ..utils import filter_by_query_param
-from .types import Page
 
 PAGE_SEARCH_FIELDS = ("content", "slug", "title")
 
 
-def resolve_page(info, page_id=None, slug=None):
-    assert page_id or slug, "No page ID or slug provided."
+def resolve_page(info, page_global_id=None, slug=None):
+    assert page_global_id or slug, "No page ID or slug provided."
```
Comment (korycins) [inline saleor/graphql/page/resolvers.py]: Wouldn't be better - `global_page_id`?

File: saleor/graphql/translations/schema.py [hunk 3]
```diff
@@ -88,3 +131,35 @@ def resolve_translations(self, info, kind, **_kwargs):
             return resolve_menu_items(info, query=None)
         elif kind == TranslatableKinds.SALE:
             return resolve_sales(info, query=None)
+
+    @permission_required("site.manage_translations")
+    def resolve_translation(self, info, id, kind, **_kwargs):
```
Comment (korycins) [inline saleor/graphql/translations/schema.py]: You don't need to do so many `return` options.
How about this:
a) 
```
if kind == TranslatableKinds.PRODUCT:
    model = Product
elif ....
...
return model.objects.filter(pk=kind_id).first()
```
or (I like this one :D )
b)
```
kinds = {TranslatableKinds.PRODUCT: Product, ..., TranslatableKinds.COLLECTION:Collection}
model = kinds[kind]
model.objects.filter(pk=kind_id).first()
```

File: saleor/graphql/schema.graphql [hunk 4]
```diff
@@ -3080,6 +3136,17 @@ type ProductPricingInfo {
   priceRangeLocalCurrency: TaxedMoneyRange
 }
 
+type ProductStrings implements Node {
```
Comment (maarcingebala) [inline saleor/graphql/schema.graphql]: `ProductTranslatableContent`

File: saleor/graphql/translations/types.py [hunk 5]
```diff
@@ -43,20 +45,108 @@ class Meta:
...
+class AttributeValueTranslatableContent(CountableDjangoObjectType):
+    translation = graphene.Field(
+        AttributeValueTranslation,
+        language_code=graphene.Argument(
+            LanguageCodeEnum,
+            description="A language code to return the translation for.",
+            required=True,
+        ),
+        description=(
+            "Returns translated Attribute Value fields " "for the given language code."
```
Comment (korycins) [inline saleor/graphql/translations/types.py]: Is it in a single line?

Comment (korycins) [reply saleor/graphql/translations/types.py]: `fields " "for the given`

File: saleor/graphql/translations/types.py [hunk 6]
```diff
@@ -43,20 +45,108 @@ class Meta:
...
+        "saleor.graphql.product.types.attributes.AttributeValue",
+        description="Represents a value of an attribute.",
+    )
+
+    class Meta:
+        model = product_models.AttributeValue
+        interfaces = [graphene.relay.Node]
+        only_fields = ["id", "name"]
+
+    def resolve_attribute_value(self, info):
```
Comment (korycins) [inline saleor/graphql/translations/types.py]: AFAIK all methods should be a statimethod:
```
    @staticmethod
    def resolve_message(root: models.CustomerEvent, _info):
```

Comment (korycins) [reply saleor/graphql/translations/types.py]: it doesn't recieve a `self. it recieves an instance of Meta.model obj

File: saleor/graphql/translations/types.py [hunk 7]
```diff
@@ -71,6 +161,42 @@ class Meta:
...
+    product = graphene.Field(
+        "saleor.graphql.product.types.products.Product",
+        description="Represents an individual item for sale in the storefront.",
+    )
+
+    class Meta:
+        model = product_models.Product
+        interfaces = [graphene.relay.Node]
+        only_fields = [
+            "description",
```
Comment (korycins) [inline saleor/graphql/translations/types.py]: I see at least 4 types that require the same `only_fields`. Maybe we could extract them to single variable?

File: saleor/graphql/translations/schema.py [hunk 8]
```diff
@@ -37,33 +51,62 @@ class Meta:
...
         TranslatableItemConnection,
         description="Returns a list of all translatable items of a given kind.",
         kind=graphene.Argument(
             TranslatableKinds, required=True, description="Kind of objects to retrieve."
         ),
     )
+    translation = graphene.Field(
+        DefaultTranslationItem,
+        id=graphene.Argument(
+            graphene.ID, description="ID of the objects to retrieve.", required=True
```
Comment (maarcingebala) [inline saleor/graphql/translations/schema.py]:
```suggestion
            graphene.ID, description="ID of the object to retrieve.", required=True
```

File: saleor/graphql/translations/schema.py [hunk 9]
```diff
@@ -37,33 +51,62 @@ class Meta:
...
+        ),
+        kind=graphene.Argument(
+            TranslatableKinds, required=True, description="Kind of objects to retrieve."
```
Comment (maarcingebala) [inline saleor/graphql/translations/schema.py]:
```suggestion
            TranslatableKinds, required=True, description="Kind of the object to retrieve."
```

**Pull request merged**